app/src/main/java/org/systers/mentorship/view/activities/SignUpActivity.kt

```diff
@@ -51,85 +65,104 @@ class SignUpActivity : BaseActivity() {
         tvTC.movementMethod = LinkMovementMethod.getInstance()
 
         btnSignUp.setOnClickListener {
+            getDetails()
 
-            name = tiName.editText?.text.toString()
-            username = tiUsername.editText?.text.toString()
-            email = tiEmail.editText?.text.toString()
-            password = tiPassword.editText?.text.toString()
-            confirmedPassword = tiConfirmPassword.editText?.text.toString()
-            needsMentoring = cbMentee.isChecked //old name but works
-            isAvailableToMentor = cbMentor.isChecked //old name but works
-            isAvailableForBoth = cbBoth.isChecked
-
-            if (validateDetails()) {
+            if (validateDetails(true)) {
                 val requestData = Register(name, username, email, password, true, needsMentoring, isAvailableToMentor)
                 signUpViewModel.register(requestData)
                 showProgressDialog(getString(R.string.signing_up))
             }
         }
+
         btnLogin.setOnClickListener {
             navigateToLoginActivity()
         }
+
         cbTC.setOnCheckedChangeListener { _, b ->
-            btnSignUp.isEnabled = b
+
+            if(! b) {
+                handler.removeCallbacks(checkInput)
+                btnSignUp.isEnabled = b
+            }
+            else checkInput.run()
         }
     }
 
     override fun onDestroy() {
         super.onDestroy()
         signUpViewModel.successful.removeObservers(this)
         signUpViewModel.successful.value = null
+
+        handler.removeCallbacks(checkInput)
+    }
+
+    private fun getDetails(){
+        name = tiName.editText?.text.toString()
+        username = tiUsername.editText?.text.toString()
+        email = tiEmail.editText?.text.toString()
+        password = tiPassword.editText?.text.toString()
+        confirmedPassword = tiConfirmPassword.editText?.text.toString()
+        needsMentoring = cbMentee.isChecked //old name but works
+        isAvailableToMentor = cbMentor.isChecked //old name but works
+        isAvailableForBoth = cbBoth.isChecked
     }
 
-    private fun validateDetails(): Boolean {
+    // b is false => checking if there are empty fields
+    // b is true => checking validity of the entries for signup
+    private fun validateDetails(b : Boolean): Boolean {
```

Comment:
can you give a proper variable name to b? I am not entirely sure what is means by reading the code. e.g.: `is...`

Reply:
'is' means :- 
if b 'is' false then I am checking if there are empty fields"
and if b is 'false' then Validation of the entries happens

This is what it meant so shall i change the variable name or let it be ??

Reply:
Did you mean "b is 'true' then Validation (...)" @CodeChamp-SS ?
Can you change `b` to be more clear, for example: `checkEmptyFields`. Something that a developer reading the code can understand the meaning right away :) Does it make sense?

Reply:
sure i'll change the var name to a more intuitive one :-)

Reply:
@isabelcosta i've made the required changes, shall i also fix the merge conflict in the pr ?

Reply:
> @isabelcosta i've made the required changes, shall i also fix the merge conflict in the pr ? I've fixed the merge conflicts but I'll have to commit all the new files added in the repo since my last commit in my pr so is that ok ?

@isabelcosta ??

Reply:
@CodeChamp-SS I'm sorry I missed this, if you updated your repo (merge the latest changes from `develop` into your branch), do push them. It should be fine, if these are changes that come from `develop`. Do it and ping me so I can review again ;)

Reply:
> @CodeChamp-SS I'm sorry I missed this, if you updated your repo (merge the latest changes from `develop` into your branch), do push them. It should be fine, if these are changes that come from `develop`. Do it and ping me so I can review again ;)

@isabelcosta done, Please review the pr again :-)